src/lib/TransactionSender.ts

```diff
@@ -43,23 +52,33 @@ export class TransactionSender {
     return this.getEthGasPrice();
   }
 
-  async getGasLimit(rawTx) {
-    const estimatedGas = await this.client.eth.estimateGas({
-      gasPrice: rawTx.gasPrice,
-      value: rawTx.value,
+  async getGasLimit(rawTx: any) {
+    const tx = {
+      from: rawTx.from,
       to: rawTx.to,
+      value: rawTx.value,
       data: rawTx.data,
-      from: rawTx.from,
-    });
-    // Gas estimation does not work correctly on RSK and after the London harfork, neither is working on Ethereum
-    // example https://etherscan.io/tx/0xd30d6cf428606e2ef3667427b9b6baecb2f4c9cbb44a0c82c735a238ec8f72fb
-    // To fix it, we decided to use a hardcoded gas estimation
-    return +estimatedGas < ESTIMATED_GAS ? ESTIMATED_GAS : +estimatedGas;
+      nonce: rawTx.nonce
+    };
+
+    const returned = await this.client.currentProvider.request(
+        {
+          method: "eth_estimateGas",
+          params: [tx],
+          jsonrpc: "2.0",
+          id: 1
+        }
+      );
+    if (returned.error) {
+      throw new Error(returned.error);
+    }
+
+    return this.client.utils.hexToNumber(returned.result as string);
   }
 
   async getEthGasPrice() {
     const chainId = await this.getChainId();
-    const gasPrice = parseInt(await this.client.eth.getGasPrice());
+    const gasPrice = Number(await this.client.eth.getGasPrice());
```

Comment:
same here maybe its better to use bigint to handle gas related stuff

Reply:
This one is used to do math with the etherscan api response that is an integer, so in this case is better to convert it.